cmake/modules/extensions.cmake

```diff
@@ -5412,8 +5412,29 @@ function(add_llext_target target_name)
   # expected `pkg_input` location. If actual POST_BUILD commands are defined,
   # they will take care of this and the default copy is replaced by a no-op.
   set(has_post_build_cmds "$<TARGET_PROPERTY:${llext_proc_target},has_post_build_cmds>")
-  set(noop_cmd ${CMAKE_COMMAND} -E true)
-  set(copy_cmd ${CMAKE_COMMAND} -E copy ${llext_lib_output} ${llext_pkg_input})
+
+  # LLEXT ELF processing for SLID imports (if enabled)
+  # Add a "hook" in the post-build commands
+  if (CONFIG_LLEXT_EXPORT_BUILTINS_BY_SLID)
+    set(noop_cmd
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/build/inject_slids_in_llext.py
+      --elf-file ${llext_lib_output}
+      -vvv
+    )
+
+    set(copy_cmd
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/build/inject_slids_in_llext.py
+      --elf-file ${llext_lib_output}
+      --output-file ${llext_pkg_input}
+      -vvv
+    )
+  else()
+    set(noop_cmd ${CMAKE_COMMAND} -E true)
+    set(copy_cmd ${CMAKE_COMMAND} -E copy ${llext_lib_output} ${llext_pkg_input})
+  endif()
```

Comment:
I don't like this as it mixes up code related to `add_llext_command` processing with SLID, executing it before any `POST_BUILD` commands run. 
I think SLID could be considered "packaging", as it does mangle the ELF names so any processing on it becomes much harder. What do you think about moving to the packaging step? 
It could be something like adding an optional `slid_cmd`:
```
  if (CONFIG_LLEXT_EXPORT_BUILTINS_BY_SLID)
    set(slid_cmd
      ... --elf-file ${llext_pkg_input} ...
    )
  else()
    set(slid_cmd ${CMAKE_COMMAND} -E true)
  endif()
```

and then later adding 
```
   COMMAND $(slid_cmd)
```
as first step to all packaging steps?

Reply:
I did think about this but couldn't think of a clean way to do it. This looks good to me though so I'll modify.

Nit: currently, `inject_slids_in_llext.py` is idempotent as it does *not* strip strings from the binary.
I did add [a check that prevents running it twice on the same LLEXT](https://github.com/mathieuchopstm/zephyr/blob/4331575936b5a8240266f69decb23ffdcf2e9e8a/scripts/build/inject_slids_in_llext.py#L100-L104), but it's an arbitrary limitation (safeguard in case something actually uses `st_value`, but AFAIK, the field is always unused for imported symbols).